### ISSUE TITLE
Custom Window Decorations on Systems != Windows 10

### ISSUE THREAD

**Issue by Crushless** [JFormDesigner/FlatLaf#242]
I consider it a bug, that FlatLaf currently only supports custom window decorations on Windows 10, despite the settings "JFrame.setDefaultLookAndFeelDecorated()" and "JDialog.setDefaultLookAndFeelDecorated()". I tried to force custom window decorations on a Test-JFrame with "setUndecorated(true)" and "getRootPane().setWindowDecorationStyle(JRootPane.FRAME)" and it works like a charm on my Plasma Desktop. The only minor bug i found, was loosing the ability do dock the JFrame to the screen borders, but I consider this as a bug in Java/AWT, and it should be fixed there and not in the LaF.

So just simply detecting the operating system is far too simple, because there are multiple windows managers with different abilities.

Basically I think, LaFs should respect user settings and only use operating system detection for default behavior.

![Screenshot_20210117_170932](https://user-images.githubusercontent.com/28498576/104848861-eacda480-58e6-11eb-9aa4-655581c49af5.png)
![Screenshot_20210117_171041](https://user-images.githubusercontent.com/28498576/104848864-ee612b80-58e6-11eb-83ad-4437c48f2b86.png)

**Comment by Bios-Marcel** [comment]
> The only minor bug i found, was loosing the ability do dock the JFrame to the screen borders

This isn't a bug, this because the docking ability is a feature of window managers decorations, which you forcefully disabled.

**Comment by DevCharly** [comment]
I agree with @Bios-Marcel 

FlatLaf custom window decorations are implemented/enabled for Windows 10 only because the tall white window title bar is annoying when using dark themes (and because Windows 10 is widely-used). On Linux there are usually dark window title bars, and they are smaller than on Windows 10.

So the only real benefit of having custom window decorations on Linux would be the embedded menu bar.

There are many window manager with different looks available for Linux.
It would be a lot of work to support even only some of them.

If you're fine with Window 10 style window border on Linux/Mac for your application, then use it.
You know how to enable it 😉
